pkg/filesystem/virtual/configuration/fuse_mount_enabled.go

```diff
@@ -45,6 +46,35 @@ func (m *fuseMount) Expose(terminationGroup program.Group, rootDirectory virtual
 	// Launch the FUSE server.
 	removeStaleMounts(m.mountPath)
 	deterministicTimestamp := uint64(filesystem.DeterministicFileModificationTimestamp.Unix())
+	mountOptions := &go_fuse.MountOptions{
+		// The name isn't strictly necessary, but is
+		// filled in to prevent runc from crashing with
+		// this error:
+		// https://github.com/opencontainers/runc/blob/v1.0.0-rc10/libcontainer/mount/mount_linux.go#L69
+		//
+		// Newer versions of runc use an improved parser
+		// that's more reliable:
+		// https://github.com/moby/sys/blob/master/mountinfo/mountinfo_linux.go
+		FsName:     m.fsName,
+		AllowOther: m.configuration.AllowOther,
+		// Speed up workloads that perform many tiny
+		// writes. This means data is only guaranteed to
+		// make it into the virtual file system after
+		// calling close()/fsync()/munmap()/msync().
+		EnableWritebackCache: true,
+	}
+
+	switch m.configuration.MountMethod {
+	case virtualpb.FUSEMountConfiguration_DIRECT:
+		mountOptions.DirectMountStrict = true
+	case virtualpb.FUSEMountConfiguration_DIRECT_AND_FUSERMOUNT:
+		mountOptions.DirectMount = true
+	case virtualpb.FUSEMountConfiguration_FUSERMOUNT:
+		// No additional options needed for FUSERMOUNT
+	default:
+		// Default to use FUSERMOUNT
```

Comment:
Please make the default fail:

return status.Error(codes.InvalidArgument, “invalid mount method”)

 And add an explicit case for FUSERMOUNT.